core/src/main/java/org/elasticsearch/search/profile/SearchProfileShardResults.java

```diff
@@ -98,15 +99,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             entry.getValue().getAggregationProfileResults().toXContent(builder, params);
             builder.endObject();
         }
-
         builder.endArray().endObject();
         return builder;
     }
 
     public static SearchProfileShardResults fromXContent(XContentParser parser) throws IOException {
         XContentParser.Token token = parser.currentToken();
         ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
-        Map<String, ProfileShardResult> searchProfileResults = new HashMap<>();
+        Map<String, ProfileShardResult> searchProfileResults = new LinkedHashMap<>();
```

Comment:
I pushed another commit that fixes the order in which the elements in the "shards" array are written out.